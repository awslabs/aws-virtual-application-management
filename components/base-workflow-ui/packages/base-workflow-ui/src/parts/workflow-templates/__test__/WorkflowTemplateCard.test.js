/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowTemplateCard from '../WorkflowTemplateCard';

describe('WorkflowTemplateCard', () => {
  const template = {
    getVersion: jest.fn(() => {
      return { id: 'test-id', v: 1, title: 'test-title', createdAt: 'Feb 1, 1966 UTC', createdBy: 'test' };
    }),
    versionNumbers: [1],
    latest: { v: 1 },
  };
  const v = 1;
  const userDisplayName = {
    isSystem: jest.fn(() => {
      return true;
    }),
    getDisplayName: jest.fn(() => {
      return 'test-displayname';
    }),
  };

  beforeEach(() => {
    jest.spyOn(global.Date, 'now').mockImplementation(() => 1626365860332);
  });

  it('WorkflowTemplateCard should renders correctly', () =>
    expect(
      renderer
        .create(
          <Provider userDisplayName={userDisplayName}>
            <WorkflowTemplateCard.WrappedComponent template={template} v={v} userDisplayName={userDisplayName} />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot());
});
