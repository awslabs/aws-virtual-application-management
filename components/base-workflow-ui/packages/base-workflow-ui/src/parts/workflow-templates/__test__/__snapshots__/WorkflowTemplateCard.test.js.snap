// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowTemplateCard WorkflowTemplateCard should renders correctly 1`] = `
Array [
  <div
    className="ui top left attached label"
    onClick={[Function]}
  >
    Template
  </div>,
  <div
    className="flex mb1"
  >
    <h3
      className="ui grey header mt0 flex-auto ellipsis"
    >
      test-title
      <div
        className="sub header fs-9"
      >
        created 
        <time
          dateTime="1966-02-01T00:00:00.000Z"
          title="Feb 1, 1966 UTC"
        >
          55 years ago
        </time>
         
        
      </div>
    </h3>
    <div
      className="ml1"
    >
      <span
        className="ellipsis pr1 fs-9 breakout color-grey"
      >
        test-id
      </span>
       
      <span
        className="bold color-grey pr2"
      >
        v
        1
      </span>
    </div>
  </div>,
  <div
    className="mt0"
  >
    <div
      className="ui pointing secondary menu"
    >
      <a
        className="active item"
        onClick={[Function]}
      >
        Description
      </a>
      <a
        className="item"
        onClick={[Function]}
      >
        Steps
      </a>
      <a
        className="item"
        onClick={[Function]}
      >
        Properties
      </a>
    </div>
    <div
      className="ui basic segment active tab"
    >
      <div
        className="animated"
      />
    </div>
  </div>,
]
`;
